pkg/agent/util/syscall/syscall_windows.go

```diff
@@ -0,0 +1,416 @@
+	SitePrefixLength                     uint32
+	Metric                               uint32
+	NlMtu                                uint32
+	Connected                            bool
+	SupportsWakeUpPatterns               bool
+	SupportsNeighborDiscovery            bool
+	SupportsRouterDiscovery              bool
+	ReachableTime                        uint32
+	TransmitOffload                      NlInterfaceOffloadRodFlags
+	ReceiveOffload                       NlInterfaceOffloadRodFlags
+	DisableDefaultRoutes                 bool
+}
+
+type RawSockAddrInet struct {
+	Family uint16
+	data   [26]byte
+}
+
+func (a *RawSockAddrInet) IP() net.IP {
+	if a == nil {
+		return net.IPv6unspecified
+	}
+	if a.Family == AF_INET {
+		addr := (*syscall.RawSockaddrInet4)(unsafe.Pointer(a))
+		return net.IPv4(addr.Addr[0], addr.Addr[1], addr.Addr[2], addr.Addr[3])
+	}
+	if a.Family == AF_INET6 {
+		addr := (*syscall.RawSockaddrInet6)(unsafe.Pointer(a))
+		return addr.Addr[:]
+	}
+	return net.IPv6unspecified
+}
+
+func (a *RawSockAddrInet) String() string {
+	return a.IP().String()
+}
+
+func NewRawSockAddrInetFromIP(ip net.IP) *RawSockAddrInet {
+	sockAddrInet := new(RawSockAddrInet)
+	if ip.To4() != nil {
+		addr, _ := netip.AddrFromSlice(ip.To4())
+		addr4 := (*windows.RawSockaddrInet4)(unsafe.Pointer(sockAddrInet))
+		addr4.Family = AF_INET
+		addr4.Addr = addr.As4()
+		addr4.Port = 0
+		addr4.Zero = [8]byte{}
+		return sockAddrInet
+	}
+	addr, _ := netip.AddrFromSlice(ip)
+	addr6 := (*windows.RawSockaddrInet6)(unsafe.Pointer(sockAddrInet))
+	addr6.Family = AF_INET6
+	addr6.Addr = addr.As16()
+	addr6.Port = 0
+	addr6.Flowinfo = 0
+	scopeId := uint32(0)
+	if z := addr.Zone(); z != "" {
+		if s, err := strconv.ParseUint(z, 10, 32); err == nil {
+			scopeId = uint32(s)
+		}
+	}
+	addr6.Scope_id = scopeId
+	return sockAddrInet
+}
+
+type AddressPrefix struct {
+	Prefix       RawSockAddrInet
+	prefixLength uint8
+	_            [2]byte // Add two bytes to keep alignment.
+}
+
+func (p *AddressPrefix) IPNet() *net.IPNet {
+	if p == nil {
+		return nil
+	}
+	sockAdrr := p.Prefix
+	if sockAdrr.Family == AF_INET {
+		return &net.IPNet{
+			IP:   (&sockAdrr).IP().To4(),
+			Mask: net.CIDRMask(int(p.prefixLength), 8*net.IPv4len),
+		}
+	}
+	if p.Prefix.Family == AF_INET6 {
+		return &net.IPNet{
+			IP:   (&sockAdrr).IP(),
+			Mask: net.CIDRMask(int(p.prefixLength), 8*net.IPv6len),
+		}
+	}
+	return nil
+}
+
+func (p *AddressPrefix) EqualsTo(dstIPnet *net.IPNet) bool {
+	if dstIPnet == nil && p == nil {
+		return true
+	} else if dstIPnet == nil || p == nil {
+		return false
+	}
+	dstOnes, dstBits := dstIPnet.Mask.Size()
+	if dstOnes != int(p.prefixLength) {
+		return false
+	}
+	if p.prefixLength == 0 {
+		return dstIPnet.IP.Equal(net.IPv4zero) || dstIPnet.IP.Equal(net.IPv6zero)
+	}
+	srcNet := p.IPNet()
+	_, srcBits := srcNet.Mask.Size()
+	if srcBits != dstBits {
+		return false
+	}
+	return srcNet.Contains(dstIPnet.IP) && dstIPnet.Contains(srcNet.IP)
```

Comment:
It could just use
```
return utilip.IPNetEqual(p.IPNet(), ipNet)
```

Reply:
Is it possible that `IP` is not the same but the two `IPNet` are the same range? In this case above change gives different results.